From: Alexander Kulak <sa-dev@rainbow.by>
Date: Mon, 5 Mar 2018 16:00:45 +0300
Subject: dh11 to 10

---
 debian/compat  | 2 +-
 debian/control | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/debian/compat b/debian/compat
index b4de394..f599e28 100644
--- a/debian/compat
+++ b/debian/compat
@@ -1 +1 @@
-11
+10
diff --git a/debian/control b/debian/control
index 37ea37b..35a263f 100644
--- a/debian/control
+++ b/debian/control
@@ -2,7 +2,7 @@ Source: alttab
 Section: x11
 Priority: optional
 Maintainer: Alexander Kulak <sa-dev@rainbow.by>
-Build-Depends: debhelper (>= 11~),
+Build-Depends: debhelper (>= 10~),
                libx11-dev,
                libxmu-dev,
                libxft-dev,
