Description: Fix some typographical errors.
Forwarded: no
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2022-08-11

--- a/doc/alttab.1.ronn
+++ b/doc/alttab.1.ronn
@@ -183,7 +183,7 @@
 
       <focus>: in multihead configuration: the monitor which shows
       largest part of currently focused window. If this part is shared
-      with other monitors, then the smallest of these monitors is choosen.
+      with other monitors, then the smallest of these monitors is chosen.
       In single head configuration: the geometry of default root window.
 
       <pointer>: the monitor which has mouse pointer, otherwise
--- a/doc/wm-setup.md
+++ b/doc/wm-setup.md
@@ -74,7 +74,7 @@
 ====
 
 ### keyboard shortcut
-Applications -> Settings -> Settings Editor (not Manager) -> Channel: xfce4-keyboard-shortcuts -> in rigth pane disable all entries of `cycle_windows_key` or `cycle_reverse_windows_key`.
+Applications -> Settings -> Settings Editor (not Manager) -> Channel: xfce4-keyboard-shortcuts -> in right pane disable all entries of `cycle_windows_key` or `cycle_reverse_windows_key`.
 
 ### startup
 Applications -> Settings -> Session and Startup -> Application Autostart -> Add
--- a/src/alttab.c
+++ b/src/alttab.c
@@ -84,7 +84,7 @@
 }
 
 //
-// initialize globals based on executable agruments and Xresources
+// initialize globals based on executable arguments and Xresources
 // return 1 if success, 0 otherwise
 // on fatal failure, calls die/exit
 //
--- a/src/gui.c
+++ b/src/gui.c
@@ -445,7 +445,7 @@
         scrdim.w = DisplayWidth(dpy, scr);
         scrdim.h = DisplayHeight(dpy, scr);
     }
-// caculate viewport.
+// calculate viewport.
 #define VPM  g.option_vp_mode
     switch (VPM) {
     case VP_SPECIFIC:
--- a/src/randr.c
+++ b/src/randr.c
@@ -249,7 +249,7 @@
         return true;
     }
     // if best cross-area is shared with some other monitor,
-    // then smallest of these monitors is choosen.
+    // then smallest of these monitors is chosen.
     smallest_2_stage_area =
         oq[best_1_stage_output].w * oq[best_1_stage_output].h;
     for (o = 0; o < no; o++) {
--- a/src/win.c
+++ b/src/win.c
@@ -669,10 +669,10 @@
         if (usec_delta < 5E5) {  // half a second
             return;
         }
-        msg(0, PREF"pulling 'prev' 0x%lx supressed\n", aw);
+        msg(0, PREF"pulling 'prev' 0x%lx suppressed\n", aw);
     }
     if (aw == g.last.to) {
-        msg(0, PREF"pulling 'to' 0x%lx supressed\n", aw);
+        msg(0, PREF"pulling 'to' 0x%lx suppressed\n", aw);
         return;
     }
 */
